lib/hotel_controller.rb

```diff
@@ -0,0 +1,80 @@
+require 'date'
+
+require_relative 'reservation.rb'
+require_relative 'room_reservation.rb'
+require_relative 'date_range.rb'
+require_relative 'room.rb'
+
+module Hotel
+  class HotelController
```

Comment:
This class is REALLY well scoped! It's letting the classes that its composed of do the heavy lifting!